JuliaInterface/src/JuliaInterface.c

```diff
@@ -36,6 +35,43 @@ static void handle_jl_exception(void)
         handle_jl_exception();                                               \
     }
 
+jl_module_t * get_module_from_string(char * name)
+{
+    // It suffices to use JULIAINTERFACE_EXCEPTION_HANDLER here, as
+    // jl_eval_string is part of the jlapi, so don't have to be wrapped in
+    // JL_TRY/JL_CATCH.
+    jl_value_t * module_value = jl_eval_string(name);
+    JULIAINTERFACE_EXCEPTION_HANDLER
+    if (!jl_is_module(module_value))
+        ErrorQuit("Not a module", 0, 0);
+    return (jl_module_t *)module_value;
+}
+
+Obj Func_InitializeGAPFFEType(){
+    JULIA_GAPFFE_type = (jl_datatype_t*)jl_new_datatype(jl_symbol("GapFFE"),
+                                                        get_module_from_string("GAP"),
+                                                        jl_any_type,
+                                                        jl_emptysvec,
+                                                        jl_svec(1,jl_symbol("ptr")),
+                                                        jl_svec(1,jl_voidpointer_type),
+                                                        0,0,1);
```

Comment:
I though some more about this, and I now think it would be better to create a [primitive type](https://docs.julialang.org/en/v1/manual/types/#Primitive-Types-1) instead, using `jl_new_primitivetype` -- because what you are stuffing into instances of this type are bits which do *not* form valid `jl_value_t` pointers, but rather are pure bits.
So roughly this:
```C
jl_value_t *name = (jl_value_t*)jl_symbol("GapFFE");
jl_module_t *module = get_module_from_string("GAP");
JULIA_GAPFFE_type = jl_new_primitivetype(name, module, jl_any_type, jl_emptysvec, 8 * sizeof(Obj));
jl_set_const(module, name, JULIA_GAPFFE_type);
```